### PR TITLE
MacOS reproducible build using Docker

### PR DESCRIPTION
**Pull request by darosior**
This finishes the work on reproducible build with the last missing cross compilation: macOS. I still need to add a README to this folder. I'll do it in this PR.

### CODE REVIEW

**Comment by darosior** [comment]
Fucking clippy

**Comment by darosior** [comment]
```
6ce1306949d07283d52d4214cea1bbf4349bbcf9a8c5824ee5f1601278fc04c1  deter_build_target/gui/x86_64-apple-darwin/release/liana-gui
ae6d2693af01afa6fd6c98c41a92fc5449d02ededd54d92d9a8a9741d79177ae  deter_build_target/x86_64-apple-darwin/release/lianad
9b17e94fb67babdae8466d256e6a9aaad7bb4369831ee69ad9a81a5ec9a2bad4  deter_build_target/gui/x86_64-pc-windows-gnu/release/liana-gui.exe
```

File: contrib/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+on Windows (the daemon isn't supported there) and both the daemon and the GUI on MacOS. The output
+will be placed in a given `TARGET_DIR` (whose default value is `deter_build_target`).
+
+### Build instructions
+
+First of all, get [Docker](https://www.docker.com/).
+
+Then get the MacOS SDK. It is required to be able to build the MacOS binaries. The `12_2` version is
+required (`Xcode_12.2.xip`). You need to download it from [Apple's
+website](https://www.docker.com/). An Apple ID and cookies enabled for the hostname are required.
```
Comment (edouardparis) [inline contrib/docker/README.md:19]: docker website

**Comment by edouardparis** [comment]
```
ae6d2693af01afa6fd6c98c41a92fc5449d02ededd54d92d9a8a9741d79177ae  deter_build_target/x86_64-apple-darwin/release/lianad
8fd59b07936dfdb31d9f139022f811ef9510b7cdb35f0661c949e38ad82a9123  deter_build_target/x86_64-apple-darwin/release/liana-cli
6ce1306949d07283d52d4214cea1bbf4349bbcf9a8c5824ee5f1601278fc04c1  deter_build_target/gui/x86_64-apple-darwin/release/liana-gui
```

**Comment by darosior** [comment]
Yay :tada:

**Comment by edouardparis** [comment]
Maybe add Xcode_12.2.xip to .gitignore

**Comment by darosior** [comment]
You don't need to have it in the repo, i think it's fine?

**Comment by darosior** [comment]
re-ACK 88c969a6a107d347ee97fbb225ae1568e98acb37

**Pull request merged**